osqp_mex.cpp

```diff
@@ -250,17 +232,17 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
         exitflag = osqp_setup(&(osqpData->solver), dataP, dataQ, dataA, dataL, dataU, dataM, dataN, settings);
         //cleanup temporary structures
         // Data
-        if (Px)       c_free(Px);
-        if (Pi)       c_free(Pi);
-        if (Pp)       c_free(Pp);
-        if (Ax)       c_free(Ax);
-        if (Ai)       c_free(Ai);
-        if (Ap)       c_free(Ap);
-        if (dataQ)    c_free(dataQ);
-        if (dataL)    c_free(dataL);
-        if (dataU)    c_free(dataU);
-        if (dataP)    c_free(dataP);
-        if (dataA)    c_free(dataA);
+        if (Px)       free(Px);
+        if (Pi)       free(Pi);
+        if (Pp)       free(Pp);
+        if (Ax)       free(Ax);
+        if (Ai)       free(Ai);
+        if (Ap)       free(Ap);
+        if (dataQ)    free(dataQ);
+        if (dataL)    free(dataL);
+        if (dataU)    free(dataU);
+        if (dataP)    free(dataP);
+        if (dataA)    free(dataA);

```

Comment:
You need to be careful to match the correct `free` function with the `malloc` function that is used. When `c_malloc` is used, `c_free` should be used. If normal `malloc` is used, then the normal `free` can be used.

In the normal library build, `c_malloc` is the same as `malloc`, however in MATLAB we map `c_malloc` to `mxMalloc` provided in the MATLAB API, which requires the use of `mxFree` to free the memory we allocate using it.